plugins/woocommerce/includes/tracks/events/class-wc-products-tracking.php

```diff
@@ -168,7 +174,15 @@ public function track_product_updated_client_side( $post ) {
 
 					window.wcTracks.recordEvent( 'product_update', properties );
 					hasRecordedEvent = true;
-				} );
+				};
+
+				if ( $( '#publish' ).length !== 0 ) {
+					$( '#publish' ).on( 'click', sendTracks);
+					return;
+				}
+
+				$( '.editor-post-publish-button' ).on( 'click', sendTracks);
```

Comment:
It appears this isn't working because the `Update`/publish button is rendered by JS and thus isn't present on the screen yet when this function is called. This is why the callback isn't triggered yet.
For the sake of keeping this simple and not the Gutenberg way, we can create a simple helper function to wait until the element exists, something like this (this waits 500ms and checks again, up to 10 times, the latter part is just out of safety):
```
function waitUntilElementExists( element, callback, limit ) {
						if ( $( element ).length > 0 ) {
						  callback();
						  return;
						}
						if ( limit < 1 ) {
						  return;
						}
						setTimeout( function() {
						    limit = limit || 10;
						    waitUntilElementExists( element, callback, --limit );
						}, 500 );
					}
					waitUntilElementExists( '.editor-post-publish-button', function() {
					  console.log('element exists');
					  $( '.editor-post-publish-button' ).on( 'click', sendTracks);
					});
```

Reply:
This is a great point. Thanks!